sycl/source/detail/pi.cpp

```diff
@@ -425,16 +428,20 @@ std::ostream &operator<<(std::ostream &Out, const DeviceBinaryProperty &P) {
   }
   Out << P.Prop->Name << "=";
 
-  switch (P.Prop->Type) {
-  case PI_PROPERTY_TYPE_UINT32:
+  if (P.Prop->Type == PI_PROPERTY_TYPE_UINT32) {
```

Comment:
NIT: original `switch` fit better here...

Reply:
Agreed, but I need to create some local variables while printing byte array specifically. I guess I could move that piece of code to a separate function and keep the `switch` here. Would that be preferrable?

Reply:
hm. not sure I understand a problem.
will this code below fit your purpose?
```cpp
switch (P.Prop->Type) {
case PI_PROPERTY_TYPE_UINT32: {
  // any locals can be created here, not visible outside this scope
  ...
  break;
...
}
...
}
```

Reply:
Ah, somehow I managed to draw a blank on that completely obvious solution. Will address.